packages/terra-slider/src/Slider.jsx

```diff
@@ -0,0 +1,122 @@
+import React, {
+  useContext, useState, useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import { injectIntl } from 'react-intl';
+import { v4 as uuidv4 } from 'uuid';
+import styles from './Slider.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+  * The label for the slider.
+  */
+  label: PropTypes.string.isRequired,
+
+  /**
+  * Minimum value in the range.
+  */
+  minimumValue: PropTypes.number.isRequired,
+
+  /**
+  * Maximum value in the range.
+  */
+  maximumValue: PropTypes.number.isRequired,
+
+  /**
+  * Default or initial value for the slider.
+  */
+  value: PropTypes.number.isRequired,
+
+  /**
+  * When true, will disable the field.
+  */
+  isDisabled: PropTypes.bool,
+
+  /**
+  * Minimum value label to display on the left side of the slider.
+  */
+  minimumLabel: PropTypes.string,
+
+  /**
+  * Maximum value label to display on the right side of the slider.
+  */
+  maximumLabel: PropTypes.string,
+
+  /**
+  * Function to trigger when user changes the input value. Sends parameter {Event} event.
+  */
+  onChange: PropTypes.func,
+
+  /**
+  * @private
+  * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
+  */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func, locale: PropTypes.string }),
+};
+
+const defaultProps = {
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Slider = props => {
+  const {
+    isDisabled,
+    intl,
+    minimumValue,
+    maximumValue,
+    label,
+    minimumLabel,
+    maximumLabel,
+  } = props;
+
+  const theme = useContext(ThemeContext);
+  const sliderClassNames = cx([
+    'slider',
+    theme.className,
+  ]);
+  const sliderRef = useRef();
+  const minLabel = minimumLabel || minimumValue;
+  const maxLabel = maximumLabel || maximumValue;
+  const visuallyHiddenTextValue = (minimumLabel && maximumLabel)
+    ? intl.formatMessage({ id: 'Terra.slider.ariaDescribedByTextWithLabels' }, {
+      minimumLabel, maximumLabel, minimumValue, maximumValue,
+    })
+    : intl.formatMessage({ id: 'Terra.slider.ariaDescribedByTextWithoutLabels' }, { minimumValue, maximumValue });
+  const descriptionId = uuidv4();
+  const [value, setValue] = useState(props.value);
```

Comment:
This should not be a state variable.  We want to consumers to be able to change the value as necessary.

Reply:
Discussed offline and concluded that the state is required to manage sliders internal updates, but a useEffect was added to update state when props change externally.